libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccountManager.java

```diff
@@ -554,4 +547,36 @@ public UserAccount getUserFromOrgAndUserId(String orgId, String userId) {
         }
         return null;
     }
+
+	/**
+	 * Attempts to refresh the access token for this user by making an API call
+	 * to the "/token" endpoint. If the call succeeds, the new token is persisted.
+	 * If the call fails and the refresh token is no longer valid, the user is logged out.
+	 * This should NOT be called from the main thread because it makes a network request.
+	 *
+	 * @param userAccount User account whose token should be refreshed. Use 'null' for current user.
+	 */
+	public synchronized void refreshToken(UserAccount userAccount) {
```

Comment:
A simple new convenience method.

Reply:
Do we need/have the same on iOS?

Reply:
There is a functionally equivalent API on iOS, but it doesn't live in SFUserAccountManager on iOS. It lives in SFSDKAuthHelper, if I'm not mistaken. Should we add one on SFUserAccountManager on iOS as well (we could simply call the existing one)? Or leave it as is?

Reply:
Thanks @bhariharan !

Reply:
Hey @bhariharan is this a synchronous network call, or is there a call back which client needs to listen to after token refresh is successful. 
Should this method return the new token or the client still needs to call the getAuthToken after this call synchronously ?

Reply:
Yes, it is a synchronous network call. I have mentioned that in the `Javadoc` for this method above. After this, you need to read the new token by updating your instance of `UserAccount` and calling `UserAccount.getAuthToken()`.